File: Day20/GCDofArray.java
import java.util.*;

public class GCDofArray {
    public int findGCD(int[] nums) {
        int min = nums[0];
        int max = nums[0];
        for (int num : nums) {
            if (num < min) min = num;
            if (num > max) max = num;
        }
        for (int i = min; i >= 1; i--) {
            if (min % i == 0 && max % i == 0) {
                return i;
            }
        }
        return 1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter numbers separated by spaces:");
        String[] parts = sc.nextLine().split(" ");
        int[] nums = Arrays.stream(parts).mapToInt(Integer::parseInt).toArray();
        System.out.println(new GCDofArray().findGCD(nums));
    }
}

File: Day20/GCDofArray.py
class Solution(object):
    def findGCD(self, nums):
        mm = min(nums)
        mx = max(nums)
        for i in range(mm, 0, -1):
            if mm % i == 0 and mx % i == 0:
                return i
        return 1

if __name__ == "__main__":
    nums = list(map(int, input("Enter numbers separated by spaces: ").split()))
    print(Solution().findGCD(nums))
